File: src/c/magma/locate/FirstLocator.c
#include <temp.h>
struct FirstLocator {
	String infix;
	struct FirstLocator new(String infix){
		struct FirstLocator this;
		this.infix = infix;
		return this;
	}
	String unwrap(){
		return this.infix;
	}
	int length(){
		return this.infix.length();
	}
	Optional<Integer> locate(){
		temp = temp;
		temp = temp;
	}
};

File: src/java/magma/locate/FirstLocator.java
package magma.locate;

import java.util.Optional;

public record FirstLocator(String slice) implements Locator {
    @Override
    public int computeLength() {
        return slice().length();
    }

    @Override
    public Optional<Integer> locate(String input) {
        final var index = input.indexOf(slice());
        if (index == -1) return Optional.empty();
        return Optional.of(index);
    }
}